File: app-mobile/src/main/java/org/jrs82/fsclock/mobile/MobileSettingsActivity.java
package org.jrs82.fsclock.mobile;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import org.jrs82.fsclock.R;
import org.jrs82.fsclock.SettingsManager;

public class MobileSettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        MobileThemeController.apply(this);
        setTheme(R.style.MobileSettingsTheme);
        super.onCreate(savedInstanceState);
        SettingsManager.get().init(getApplicationContext());

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(android.R.id.content, new MobileSettingsFragment())
                    .commit();
        }
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle(R.string.mobile_settings_title);
        }
    }
}

File: app-mobile/src/main/java/org/jrs82/fsclock/mobile/MobileSettingsActivity.kt
package org.jrs82.fsclock.mobile

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import org.jrs82.fsclock.SettingsManager

/**
 * Asetus-Activity. UI on Compose ([SettingsScreen]); pysyy AppCompatActivityna jotta
 * [MobileThemeController]in night mode -pakotus (light/dark/system) toimii ja heijastuu
 * Composen isSystemInDarkTheme():iin. Teema manifestissa = MobileComposeTheme (NoActionBar).
 */
class MobileSettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        MobileThemeController.apply(this)
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
        SettingsManager.get().init(applicationContext)
        setContent {
            ArkikeskusTheme {
                SettingsScreen(onBack = { finish() })
            }
        }
    }
}
